e2e/fixtures/fdb_data_loader.go

```diff
@@ -0,0 +1,197 @@
+            - "--copy-library"
+            - "{{ .FDBVersion.Compact }}"
+            - "--output-dir"
+            - "/var/output-files/primary" # Note that we use primary as the subdirectory rather than specifying the FoundationDB version like we did in the other examples.
+            - "--init-mode"
+          volumeMounts:
+            - name: fdb-libs
+              mountPath: /var/output-files
+        {{ end }}
+        {{ end }}
+      restartPolicy: Never
+      volumes:
+        - name: config-map
+          configMap:
+            name: {{ .ClusterName }}-config
+            items:
+              - key: cluster-file
+                path: fdb.cluster
+        - name: fdb-libs
+          emptyDir: {}
+        - name: fdb-certs
+          secret:
+            secretName: {{ .SecretName }}`
+)
+
+// dataLoaderConfig represents the configuration of the Dataloader Job.
+type dataLoaderConfig struct {
+	// Image represents the data loader image that should be used in the Job.
+	Image string
+	// SidecarVersions represents the sidecar configurations for different FoundationDB versions.
+	SidecarVersions []SidecarConfig
+	// Namespace represents the namespace for the Deployment and all associated resources
+	Namespace string
+	// ClusterName the name of the cluster to load data into.
+	ClusterName string
+	// SecretName represents the Kubernetes secret that contains the certificates for communicating with the FoundationDB
+	// cluster.
+	SecretName string
+}
+
+func (factory *Factory) getDataLoaderConfig(cluster *FdbCluster) *dataLoaderConfig {
+	return &dataLoaderConfig{
+		Image:           factory.GetDataLoaderImage(),
+		Namespace:       cluster.Namespace(),
+		SidecarVersions: factory.GetSidecarConfigs(),
+		ClusterName:     cluster.Name(),
+		SecretName:      factory.GetSecretName(),
+	}
+}
+
+// CreateDataLoaderIfAbsent will create the data loader for the provided cluster and load some random data into the cluster.
+func (factory *Factory) CreateDataLoaderIfAbsent(cluster *FdbCluster) {
+	if !factory.options.enableDataLoading {
+		return
+	}
+
+	t, err := template.New("dataLoaderJob").Parse(dataLoaderJob)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	buf := bytes.Buffer{}
+	gomega.Expect(t.Execute(&buf, factory.getDataLoaderConfig(cluster))).NotTo(gomega.HaveOccurred())
+	decoder := yamlutil.NewYAMLOrJSONDecoder(&buf, 100000)
+	for {
+		var rawObj runtime.RawExtension
+		err := decoder.Decode(&rawObj)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		obj, _, err := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).
+			Decode(rawObj.Raw, nil, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		unstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		unstructuredObj := &unstructured.Unstructured{Object: unstructuredMap}
```

Comment:
Where is the client actually loading the data into the key-value store?

Reply:
Here: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/sample-apps/data-loader/app.py#L23 we use the sample data loader app.